### PR TITLE
feat(pagination): added 'infinite lists' pagination options

### LINKED ISSUE #13188: ajax load more comments in comments list

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #13188

### CODE REVIEW

File: views/default/page/components/list/ajax-append.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			},
+			success: function(result) {
+				var id_selector = '#' + $target.attr('id');
+
+				$new_html = $(result).find(id_selector).addBack(id_selector);
+				if ($new_html.length) {
+					// update history
+					window.history.pushState('', '', href);
+
+					// determine position
```
Comment (jdalsem) [inline views/default/page/components/list/ajax-append.js:32]: could be removed

File: views/default/page/components/list/ajax-append-auto.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	    var elemTop = $elem.offset().top;
+	    var elemBottom = elemTop + $elem.height();
+
+	    if ((elemBottom <= docViewBottom) && (elemTop >= docViewTop)) {
+	    	$elem.click();
+	    }
+	};
+	
+	$(document).off('scroll.ajax-append-auto');
```
Comment (jdalsem) [inline views/default/page/components/list/ajax-append-auto.js:38]: shoudl not be necessary...

File: views/default/page/components/list/ajax-append-auto.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		};
+	};
+	
+	var checkScroll = function() {
+		var $elem = $('.elgg-list-container-ajax-append-auto .elgg-pagination-next > a:visible');
+		if (!$elem.length) {
+			return;
+		}
+		
+		var docViewTop = $(window).scrollTop();
```
Comment (jdalsem) [inline views/default/page/components/list/ajax-append-auto.js:27]: indentation issues?

File: views/default/page/components/list/ajax-append-auto.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+			var later = function() {
+				debounceTimeout = null;
+				if (!immediate) func.apply(context, args);
+			};
+			var callNow = immediate && !timeout;
+			clearTimeout(debounceTimeout);
+			debounceTimeout = setTimeout(later, wait);
+			if (callNow) func.apply(context, args);
+		};
+	};
```
Comment (jeabakker) [inline views/default/page/components/list/ajax-append-auto.js:19]: maybe explain why this function exists

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: views/default/page/components/list/ajax-append-auto.js (hunk 4 above)
Comment (jeabakker) [reply views/default/page/components/list/ajax-append-auto.js:19]: https://davidwalsh.name/javascript-debounce-function

File: views/default/page/components/list/ajax-append-auto.js (hunk 2 above)
Comment (jdalsem) [reply views/default/page/components/list/ajax-append-auto.js:38]: it is

**Review by jeabakker — APPROVED** [review]

**Pull request merged**